src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -730,6 +733,36 @@ SubCmdProgram::execute(const SubCmdOptions& _options) const
     throw xrt_core::error("uuid does not match BLP");
   }
 
+  // -- process "user" option ---------------------------------------
+  if(!xclbin.empty()) {
+    XBU::verbose(boost::str(boost::format("  xclbin: %s") % xclbin));
+    //only 1 card and name
+    if(deviceCollection.size() > 1)
+      throw xrt_core::error("Please specify a single device");
+    auto dev = deviceCollection.front();
+
+    std::ifstream stream(xclbin, std::ios::binary);
+    if (!stream)
+      throw xrt_core::error(boost::str(boost::format("Could not open %s for reding") % xclbin));
+
+    stream.seekg(0,stream.end);
+    ssize_t size = stream.tellg();
+    stream.seekg(0,stream.beg);
+
+    std::vector<char> raw(size);
+    stream.read(raw.data(), size);
+    std::string v(raw.data(),raw.data()+7);
+    if (v != "xclbin2")
+      throw xrt_core::error(boost::str(boost::format("Bad binary version '%s'") % v));
```

Comment:
The Error message while valid is:

1. Not very helpful.
2. Will produce garbage if pointing to a binary file.

Q: Why is this check being done here as opposed to in the "load_xclbin" method?

Reply:
What would be a more helpful error in your opinion? `v` is just the xclbin magic string, i.e., `xclbin2` so even if it produces garbage, it will only print a few beginning characters which might be helpful